iocore/cache/Cache.cc

```diff
@@ -660,14 +660,13 @@ CacheProcessor::start_internal(int flags)
           }
```

Comment:
The `diskok` was initialized outside of the loop. At here, the value of diskok is unpredictable.

Reply:
Since you are touching it, maybe clean it up all the way:
```
diff --git a/iocore/cache/Cache.cc b/iocore/cache/Cache.cc
index e464e191e..0610c3d2c 100644
--- a/iocore/cache/Cache.cc
+++ b/iocore/cache/Cache.cc
@@ -607,7 +607,6 @@ CacheProcessor::start_internal(int flags)
   fix                  = !!(flags & PROCESSOR_FIX);
   check                = (flags & PROCESSOR_CHECK) != 0;
   start_done           = 0;
-  int diskok           = 1;
   Span *sd;
 
   /* read the config file and create the data structures corresponding
@@ -652,25 +651,26 @@ CacheProcessor::start_internal(int flags)
     }
 
     if (fd >= 0) {
+      bool diskok = false;
+
       if (!sd->file_pathname) {
         if (!check) {
           if (ftruncate(fd, blocks * STORE_BLOCK_SIZE) < 0) {
             Warning("unable to truncate cache file '%s' to %" PRId64 " blocks", path, blocks);
-            diskok = 0;
           }
         } else { // read-only mode checks
           struct stat sbuf;
-          diskok = 0;
           if (-1 == fstat(fd, &sbuf)) {
             fprintf(stderr, "Failed to stat cache file for directory %s\n", path);
           } else if (blocks != sbuf.st_size / STORE_BLOCK_SIZE) {
             fprintf(stderr, "Cache file for directory %s is %" PRId64 " bytes, expected %" PRId64 "\n", path, sbuf.st_size,
                     blocks * static_cast<int64_t>(STORE_BLOCK_SIZE));
           } else {
-            diskok = 1;
+            diskok = true;
           }
         }
       }
+
       if (diskok) {
         int sector_size = sd->hw_sector_size;

```

Reply:
change the type to bool.